source/entities/Building.cpp

```diff
@@ -31,6 +32,57 @@ const Ogre::Vector3 SCALE(RenderManager::BLENDER_UNITS_PER_OGRE_UNIT,
         RenderManager::BLENDER_UNITS_PER_OGRE_UNIT,
         RenderManager::BLENDER_UNITS_PER_OGRE_UNIT);
 
+Building::~Building()
+{
+    for(std::pair<Tile* const, TileData*>& p : mTileData)
+    {
+        delete p.second;
+    }
+    mTileData.clear();
```

Comment:
There is not really a need to call clear if the object is being destroyed anyway.